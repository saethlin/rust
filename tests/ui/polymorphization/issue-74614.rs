//@ compile-flags:-Zpolymorphize=on
//@ build-pass
//@ dont-check-compiler-stderr

fn test<T>() {
    std::mem::size_of::<T>();
}

pub fn foo<T>(_: T) -> &'static fn() {
    &(test::<T> as fn())
}

fn outer<T>() {
    foo(|| ());
}

fn main() {
    outer::<u8>();
}
