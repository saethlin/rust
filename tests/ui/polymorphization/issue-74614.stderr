 WARN rustc_codegen_ssa::mir::locals Unexpected initial operand type:
 expected Closure(DefId(0:9 ~ issue_74614[b919]::outer::{closure#0}), [Closure(DefId(0:9 ~ issue_74614[b919]::outer::{closure#0}), [T/#0, i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]), i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]),
 found    Closure(DefId(0:9 ~ issue_74614[b919]::outer::{closure#0}), [T/#0, i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]).
 See <https://github.com/rust-lang/rust/issues/114858>.
